### DISCUSSION TITLE
Some 5.6.0 questions

### DISCUSSION THREAD

**Discussion by rschmunk** [Unidata/netcdf-java#1370] [Questions and Answers] [answered]
Im trying to work with the 5.6.0 release and have run into a couple things. Nothing I'd call a bug yet; they may just be me doing something wrong.

The size of `netcdfAll-5.6.0.jar` is about 30 MB (built with the S3 and Zarr modules included) while a `netcdfAll-5.5.4-SNAPSHOT.jar` built from a snapshot of a couple weeks ago is about 22 MB. Is there any explanation for the size jump? Some third party library that's been added recently?

Also, does the gradle build process use a different, more efficient zip compression tool to create the final jar file? In order to get my app to notarize for use on a Mac, I have found that I have to unjar the netcdfAll build and then re-jar it, which I do by just calling jar from the command-line. Otherwise, the Apple notarization tool complains that some x86_64 native code in the build is not signed. However, the re-jarred library jumps in size, almost double what it was before. Note, this is not new to 5.6.0.

Finally, although the jdom2 classes appears to be in the netcdfAll build, my app is complaining that `XMLReaderJDOMFactory` is not present. This occurs before any netCDF-Java code is called. My app uses jdom2 to parse a preferences file, and I've been relying on the jdom2 library being included in netcdfAll. Since the class is included in the netcdfAll build, this one is mystifying me.

**Comment by haileyajohnson** [comment]
The netcdfAll-5.6.0.jar I see on our server is 20.9 MB and 21.9 MB when downloaded from our downloads page... so you're seeing 30MB before re-jarring and then almost 60MB after?

Gradle uses the deflate (zip) algorithm to create the jars, my guess is you have a different deflate level set, but I can't for the life of me find what the default for gradle is. 

Skipping ahead to your jdom2 issue: we upgraded the version of the jdom library we use due to security issues, though that would have been present in the latest snapshot as well... `XMLReaderJDOMFactory is deprecated however, they recommend `SAXParserFactory` instead [(](https://docs.oracle.com/javase/9/docs/api/org/xml/sax/helpers/XMLReaderFactory.html), which is what we use in our code.

**Comment by rschmunk** [comment, marked as answer]
@haileyajohnson, If I do a `./gradlew builtNetcdfall` without the S3 or Zarr code, then I get a 20.9 MB jar. Which okay, I just discovered that the 5.5.4 snapshot I have on this particular machine was not currently configured to include those modules when building. Further investigation reveals that the S3 module is responsible for about 75-80% of the size jump I was seeing and the Zarr module the rest.

And... I just found a serious flaw in my build-unjar-rejar script on this particular machine which is the source of the mega-bloat. Fixing that, I find that the new jar only goes up maybe 2% in size rather than doubling.

And... third times a charm, my fault again. My app is now seeing the `XMLReaderJDOMFactory` class in the netdfAll-5.6.0 jar. It looks like there was a typo in my app build script and the resulting classpath was messed up. I'll have to revisit my XML-preferences code (it's pretty old) to see if updating to `SAXParserFactory` makes sense.
